phases/01-math-foundations/12-tensor-operations/code/notes.py

```
@@ -0,0 +1,29 @@
+class Tensor:
+    def __init__(self, data, shape=None):
+        if isinstance(data, (list, tuple)):
+            self._data, self._shape = self._flatten_nested(data)
+        elif isinstance(data, np.ndarray):
```

Ruff (0.16.1)

[error] Undefined name `np`

(F821)